src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPlugin.groovy

```diff
@@ -98,42 +117,59 @@ class DockerJavaApplicationPlugin implements Plugin<Project> {
             description = "Copies the distribution resources to a temporary directory for image creation."
             dependsOn installTask
             from { installTask.destinationDir.parentFile }
-            into { createDockerfileTask.destFile.parentFile }
+            into { createDockerfileTask.destFile.get().asFile.parentFile }
             exclude "**/lib/${jarTask.archiveName}"
             into("app-lib") {
                 from jarTask
             }
         } as Sync
     }
 
-    private String determineEntryPoint(Project project, Sync installTask) {
-        "/${installTask.destinationDir.name}/bin/${project.applicationName}".toString()
+    private Provider<List<String>> determineEntryPoint(Project project, Sync installTask) {
+        project.provider(new Callable<List<String>>() {
+            @Override
+            List<String> call() throws Exception {
+                ["/${installTask.destinationDir.name}/bin/${project.applicationName}".toString()]
+            }
+        })
     }
 
     private DockerBuildImage createBuildImageTask(Project project, Dockerfile createDockerfileTask, DockerJavaApplication dockerJavaApplication) {
         project.task(BUILD_IMAGE_TASK_NAME, type: DockerBuildImage) {
             description = 'Builds the Docker image for the Java application.'
             dependsOn createDockerfileTask
-            conventionMapping.inputDir = { createDockerfileTask.destFile.parentFile }
-            conventionMapping.tag = { determineImageTag(project, dockerJavaApplication) }
+            inputDir = project.provider(new Callable<Directory>() {
+                @Override
+                Directory call() throws Exception {
+                    DirectoryProperty inputDirProperty = project.layout.directoryProperty()
+                    inputDirProperty.set(createDockerfileTask.destFile.get().asFile.parentFile)
+                    inputDirProperty.get()
```

Comment:
Uhhhhh.... What? Creating a provider just to return it's value? Are you sure this is what you want to be doing here??